### PR TITLE
Cms/new recurring background jobs

### PR DESCRIPTION
**Pull request by andrewmckaskill**
## Description

Updated the documentation around scheduling background jobs to reflect:
-  the new IRecurringBackgroundJob interface
- the addition of notifications
- more examples

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [X] New content
* [X] Updates related to a new version
* [ ] Other

## Product & version (if relevant)

Umbraco CMS v13

## Deadline (if relevant)

Content should be published alongside v13 release (although as the RC is now available it could be published straight away)

### CODE REVIEW

**Comment by andrewmckaskill** [comment]
The final vale issue has detected the work "azure" in lower case inside a link.

```md
-   MainDom - The `MainDom` lock ensures that only one instance of Umbraco is running at a time on a given machine. This ensures the integrity of certain files used by Umbraco. See [../fundamentals/setup/server-setup/load-balancing/azure-web-apps#host-synchronization](Host Synchronization) for more details.
```

Is it possible to ignore that warning?

**Comment by jonat123** [comment]
@andrewmckaskill Thanks for the PR 😊
We will make sure to review it as soon as possible!

And yes, the warning can just be ignored 😄

File: 13/umbraco-cms/reference/scheduling.md [hunk 1]
```diff
@@ -83,83 +222,73 @@ public class CleanUpYourRoom : RecurringHostedServiceBase
...
-{
-    case ServerRole.Subscriber:
-        _logger.LogDebug("Does not run on subscriber servers.");
-        return Task.CompletedTask; // We return Task.CompletedTask to try again as the server role may change!
-    case ServerRole.Unknown:
-        _logger.LogDebug("Does not run on servers with unknown role.");
-        return Task.CompletedTask; // We return Task.CompletedTask to try again as the server role may change! 
-}
-```
+This notification is there to support low-level debugging of background jobs to ascertain why they are / are not running. As the runtime checks include runtime state readiness, this event may be triggered during the install phase. Any notification handlers associated with this notification should **also** conduct their own checks before relying on Umbraco services, including database access.
```
Comment (alina-tincas) [inline 13/umbraco-cms/reference/scheduling.md:280]:
```suggestion
This notification is there to support low-level debugging of background jobs to ascertain why they are/aren't running. As the runtime checks include runtime state readiness, this event may be triggered during the install phase. Any notification handlers associated with this notification should **also** conduct their own checks before relying on Umbraco services, including database access.
```

File: 13/umbraco-cms/reference/scheduling.md [hunk 2]
```diff
@@ -83,83 +222,73 @@ public class CleanUpYourRoom : RecurringHostedServiceBase
...
+
+### Executing / Executed / Failed
```
Comment (alina-tincas) [inline 13/umbraco-cms/reference/scheduling.md:282]:
```suggestion
### Executing/Executed/Failed
```

File: 13/umbraco-cms/reference/scheduling.md [hunk 3]
```diff
@@ -83,83 +222,73 @@ public class CleanUpYourRoom : RecurringHostedServiceBase
...
+
+These notifications will be triggered in pairs depending on the success / failure of the job itself. The executing notification is triggered before the job is run. The executed notification is triggered after the job completes. The failed notification is triggered from the catch block if an exception is thrown.
```
Comment (alina-tincas) [inline 13/umbraco-cms/reference/scheduling.md:284]:
```suggestion
These notifications will be triggered in pairs depending on the success/failure of the job itself. 
- The executing notification is triggered before the job is run. 
- The executed notification is triggered after the job is completed. 
- The failed notification is triggered from the catch block if an exception is thrown.
```

**Comment by alina-tincas** [comment]
Hi @andrewmckaskill thank you for the PR! #h5YR 

I have also noticed that it is based on the CMS PR https://github.com/umbraco/Umbraco-CMS/pull/14291 which is good to know 💪 

The reason why there was a vale error for azure is because the links was created wrongly. Instead of [../fundamentals/setup/server-setup/load-balancing/azure-web-apps#host-synchronization](Host Synchronization) it should be [Host Synchronization](../fundamentals/setup/server-setup/load-balancing/azure-web-apps#host-synchronization). I have changed this and now there is no vale error 😁 I have also made some small wording changes to make the content readable.

I have also tested out the codes, where the Example with dependency injection works for me, but not the Complex example as I get the following:
image

1. Can you please look into the above complex example?
2. Can you please add a short description/explanation before the **minimal example** of what are the examples about? Ideas: "The following examples create a background job that empties the content's recycle bin every 60 minutes."
3. As there is a new IRecurringBackgroundJob interface, then I was wondering if you would be up for looking into the following article and see if anything needs to be changed here as well: https://docs.umbraco.com/umbraco-cms/reference/notifications/creating-and-publishing-notifications? Otherwise we can look into updating this article. If this is the case, would this be a matter of just changing anywhere where it is mentioned:
- from using `Umbraco.Cms.Infrastructure.HostedServices;` change to` using Umbraco.Cms.Infrastructure.BackgroundJobs;`
- from `RecurringHostedServiceBase `interface to `IRecurringBackgroundJob `?

Looking forward to hearing back from you on the above 😊

**Comment by alina-tincas** [comment]
Hi @andrewmckaskill I was wondering if you have seen my latest message and if you are able to look into the complex example? 😊 

Looking forward to hearing back from you soon!

**Comment by andrewmckaskill** [comment]
Hi Alina - apologies. I missed the notification. I'll try and sort out the examples today.

**Comment by alina-tincas** [comment]
Sounds good, thank you @andrewmckaskill 🙌

**Comment by andrewmckaskill** [comment]
Hi @alina-tincas those updates are done now as requested!

**Review by alina-tincas — APPROVED** [review]

**Comment by alina-tincas** [comment]
Hi @andrewmckaskill thank you for making all those changes #h5YR 😊 🙌 

Merging those changes in!

File: 13/umbraco-cms/reference/scheduling.md [hunk 4]
```diff
@@ -1,63 +1,207 @@
...
+        // Remember to complete the scope when done.
+        scope.Complete();
+        return Task.CompletedTask;
+    }
+}
+```
+
+## Complex example
+
+The complex example builds on the previous one by injecting additional services, including a logger to log error messages, a profiler to capture timings, and an `IServerRoleAccessor` to log the current server role. Additionally it injects an `IOptionsMonitor` to allow the period to be updated while the server is running, and demonstrates how to trigger the `PeriodChanged` event to signal the job's host.
```
Comment (alina-tincas) [inline 13/umbraco-cms/reference/scheduling.md:146]:
```suggestion
The complex example builds on the previous one by injecting additional services. It includes a logger to log error messages, a profiler to capture timings, and an `IServerRoleAccessor` to log the current server role.  Additionally, it injects an `IOptionsMonitor` to allow the period to be updated while the server is running. It also demonstrates how to trigger the `PeriodChanged` event to signal the job's host.
```

**Review by alina-tincas — APPROVED** [review]
fixing vale errors after resolving conflicts

**Pull request merged**